### ISSUE TITLE
Occasionally freeze when moving the mouse in the editor

### ISSUE THREAD

**Issue by Rindbee** [godotengine/godot#69942]
### Godot version

f692b4768

### System information

Linux mint 21, forward_plus ,NVIDIA GeForce GTX 1050 Ti

### Issue description

I occasionally get stuck when moving the mouse in the editor. I have encountered it many times so far, but I have not explored the specific cause. Cannot use the screenshot tool to take a screenshot after being stuck.

Recently I tried to reproduce it by just moving the mouse, and I only succeeded once, but still can't find any reproducible steps. Might be related to the tooltip popup. When the popup does not disappear with the mouse movement, it means that it has been stuck.

I use i3wm, prone to this problem in the following situations. 

1. When switching workspaces, you may need to stay in other workspaces for a while; 
2. Or let the editor display on the screen, leave the computer for a while and use the mouse to move again.

### Steps to reproduce

Sorry, this issue came up by accident and I'm not sure of the exact steps to reproduce.

### Minimal reproduction project

N/A

**Comment by timothyqiu** [comment]
I was just about to report the same issue :+1: 

I use Arch Linux and was using a `3.x` custom build. I once succeeded reproducing the freeze by attaching a new script, typing some export variable declaration, and randomly clicking the mouse. But those steps no longer work now.

When I was able to reproduce the freeze with gdb on, interrupting the process shows that the main thread is stuck at a pthread mutex lock inside the `XSetICValues()` in `OS_X11::set_ime_position()` (called by `TextEdit::_notification()`). I'm not sure if this is related to the freeze.

https://github.com/godotengine/godot/blob/f128170a1878758e27ee4c9ded7e74ebcde210b8/platform/x11/os_x11.cpp#L828-L848

**Comment by lawnjelly** [comment]
Notable all reports are on linux. Has this occurred before or is it recent?

Most recent PR is #69943. (If this is occurring on 3.x and master).

Just looking at that PR, I can see a possible problem:

```
void JoypadLinux::monitor_joypads() {
	while (!monitor_joypads_exit.is_set()) {
		DIR *input_directory;
		input_directory = opendir("/dev/input");
		if (input_directory) {
			struct dirent *current;
			char fname[64];
			while ((current = readdir(input_directory)) != nullptr) {
				if (strncmp(current->d_name, "event", 5) != 0) {
					continue;
				}
				sprintf(fname, "/dev/input/%.*s", 16, current->d_name);
				if (attached_devices.find(fname) == -1) {
					open_joypad(fname);
				}
			}
		}
		closedir(input_directory);
		usleep(1000000); // 1s
	}
REMOVED	usleep(1000000); // 1s
}
```
How often is `monitor_joypads()` called? Should there be a sleep in _both_ locations?

**Comment by KaziiTheAvali** [comment]
i have this problem too. the current and only seaming work around is using 3.2.3 as that is the latest working version. 3.3.x breaks in the same way as 3.5.1

**Comment by artyommironov** [comment]
Void Linux, Wayland, Sway 1.8.2. Godot 3.5.1 constantly freezes in the editor after few minutes of work. As @timothyqiu mentioned `set_ime_position()` I thought maybe IME is causing this issue. I had `fcitx5` running as IME, when I stop it Godot works without freezing.

**Comment by lawnjelly** [comment]
BTW for anyone getting this, it would be really helpful to know how far back in terms of versions this freezing happens, because it might help pinpoint which change might be responsible (i.e. if you download some old builds 3.4, 3.3 etc and see if you can work out how far back this was happening, or when it started).

**Comment by artyommironov** [comment]
> it would be really helpful to know how far back in terms of versions this freezing happens

For me it seems:
- 3.2.3 no issue
- 3.2.4-beta1 has issue

**Comment by lawnjelly** [comment]
> [This commit](https://github.com/godotengine/godot/commit/f725d9cb733e103c91d889777b78c5a646095658) happened between these versions and looks related.

That indeed looks likely as it looks like the `events_mutex` might have been responsible in @timothyqiu 's bit of code, and that was introduced in #42341 .

**Comment by Zireael07** [comment]
IME stands for Input Method Editors and it is mostly used in Asian languages. (Arabic uses a RTL keyboard layout, which does not need an IME)

**Comment by artyommironov** [comment]
> By any chance are all you guys with this problem using non-english language? Or has anyone reproducing this with english input?

@lawnjelly I use IME (fcitx5) for Japanese input ocassionally, but I only use English layout in Godot. In fact, to reproduce the freeze I don't even need to input anything, just start selecting text in the editor with mouse and moving the mouse cursor around. I'm not sure that IMEs others use, but this might be partially a fcitx5 issue, IME support in Wayland is not perfect and fcitx5 was the only IME that actually worked for me. Maybe Godot makes some expectations about IME that doesn't happen since it's not so stable in Wayland.

**Comment by Zireael07** [comment]
> when we select text it seems got auto-copied to PRIMARY clipboard

That's a system thing not a Godot thing

**Comment by artyommironov** [comment]
> That's a system thing not a Godot thing

True, I was confused by PRIMARY mentioned in `set_clipboard`, but after some checking it turns out copying is not happending during selection, actually it happens when paste event triggered somewhere.

**Comment by Rindbee** [comment]
It seems that a thread is reserved when compiling (```scons -j$((`nproc` - 1))```), and it is easier to reproduce this issue in the editor. I'm stuck in the following situations in a7276f1ce:
1. After creating a new project, it is stuck in the import progress bar;
2. After the OptionButton pops up the options;
3. When renaming the node name.
